src/app/rpmostree-override-builtins.cxx

```diff
@@ -89,6 +91,47 @@ handle_override (RPMOSTreeSysroot  *sysroot_proxy,
   g_autoptr(GVariant) options = g_variant_ref_sink (g_variant_dict_end (&dict));
 
   g_autoptr(GVariant) previous_deployment = rpmostree_os_dup_default_deployment (os_proxy);
+//  g_autofree char *transaction_address = NULL;                                          
+//  if (!rpmostree_os_call_refresh_md_sync(os_proxy,&transaction_address,cancellable,error))
+//    return FALSE;
+  if (override_replace && opt_ex_pin_from_repos)
+    {
+      g_autoptr(GPtrArray) override_replace_final = NULL;
+      GRegex *is_http = g_regex_new("/^http:///", (GRegexCompileFlags)0, 
+                                    (GRegexMatchFlags)0, error);
+      GRegex *is_https = g_regex_new("/^https:///", (GRegexCompileFlags)0, 
+                                     (GRegexMatchFlags)0, error);
+      GRegex *ends_with_rpm = g_regex_new("/.rpm$/", (GRegexCompileFlags)0, 
+                                          (GRegexMatchFlags)0, error);
```

Comment:
You can use `g_str_has_prefix` and `g_str_has_suffix` instead. Or if you'd like, add a new Rust function to do this so it's colocated with the client_handle_fd function which does something similar.